internal/auth0/organization/resource_members.go

```diff
@@ -0,0 +1,203 @@
+			},
+			"members": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: "Add user ID(s) directly from the tenant to become members of the organization.",
+			},
+		},
+	}
+}
+
+func createOrganizationMembers(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	alreadyMembers, err := api.Organization.Members(organizationID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID)
+
+	membersToAdd := *value.Strings(data.GetRawConfig().GetAttr("members"))
+
+	if diagnostics := guardAgainstErasingUnwantedMembers(
+		organizationID,
+		alreadyMembers.Members,
+		membersToAdd,
+	); diagnostics.HasError() {
+		data.SetId("")
+		return diagnostics
+	}
+
+	if err := api.Organization.AddMembers(organizationID, membersToAdd); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return readOrganizationMembers(ctx, data, meta)
+}
+
+func readOrganizationMembers(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	members, err := api.Organization.Members(data.Id())
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	result := multierror.Append(
+		data.Set("organization_id", data.Id()),
+		data.Set("members", flattenOrganizationMembers(members.Members)),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func updateOrganizationMembers(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	toAdd, toRemove := value.Difference(data, "members")
+
+	removeMembers := make([]string, 0)
+	for _, member := range toRemove {
+		removeMembers = append(removeMembers, member.(string))
+	}
+
+	if len(removeMembers) > 0 {
+		if err := api.Organization.DeleteMember(organizationID, removeMembers); err != nil {
+			if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+				data.SetId("")
+				return nil
+			}
+
+			return diag.FromErr(err)
+		}
+	}
+
+	addMembers := make([]string, 0)
+	for _, member := range toAdd {
+		addMembers = append(addMembers, member.(string))
+	}
+
+	if len(addMembers) > 0 {
+		if err := api.Organization.AddMembers(organizationID, addMembers); err != nil {
+			return diag.FromErr(err)
+		}
```

Comment:
I know you've been wanting to avoid redundant 404 checks, but purely additive changes that result in 404s would not get appropriately handled here as it does above with the delete.

Reply:
Good point 👍🏻 , updated in [`9d1583d` (#614)](https://github.com/auth0/terraform-provider-auth0/pull/614/commits/9d1583d60fcd55406bd65a22b4bab6e63384b907)